# Fix for Linux kernel 6.12 (https://github.com/atar-axis/xpadneo/issues/498)
# Patch based on kakra (Kai Kakrow) fixes:
# - https://github.com/atar-axis/xpadneo/commit/4bfe0a1c356a0fc475a014d835ba97a710c624b8
# - https://github.com/kakra/xpadneo/commit/746de0f2d1cc334f5f03ed3994555a233b0a46c1

diff --git a/hid-xpadneo/src/hid-xpadneo.c b/hid-xpadneo/src/hid-xpadneo.c
index 1a25631..c70b7cf 100644
--- a/hid-xpadneo/src/hid-xpadneo.c
+++ b/hid-xpadneo/src/hid-xpadneo.c
@@ -10,6 +10,7 @@
 
 #include <linux/delay.h>
 #include <linux/module.h>
+#include <linux/version.h>
 
 #include "xpadneo.h"
 
@@ -712,7 +713,11 @@ static int xpadneo_input_mapping(struct hid_device *hdev, struct hid_input *hi,
 	return MAP_AUTO;
 }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6,12,0)
 static u8 *xpadneo_report_fixup(struct hid_device *hdev, u8 *rdesc, unsigned int *rsize)
+#else
+static const u8 *xpadneo_report_fixup(struct hid_device *hdev, u8 *rdesc, unsigned int *rsize)
+#endif
 {
 	struct xpadneo_devdata *xdata = hid_get_drvdata(hdev);
 
